<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1857</width>
    <height>1032</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="change_point">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>790</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>change point</string>
    </property>
   </widget>
   <widget class="QPushButton" name="open_source">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>740</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>open source</string>
    </property>
   </widget>
   <widget class="QPushButton" name="insert_point">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>740</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>insert point</string>
    </property>
   </widget>
   <widget class="QPushButton" name="delete_point">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>790</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>delete point</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Start">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>770</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>start</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>1520</x>
      <y>20</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>MODE</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>1460</x>
      <y>70</y>
      <width>181</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>TransformationEstimate</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>1500</x>
      <y>310</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Correspondence</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>1500</x>
      <y>170</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Parameter</string>
    </property>
   </widget>
   <widget class="QComboBox" name="Paraments">
    <property name="geometry">
     <rect>
      <x>1440</x>
      <y>210</y>
      <width>91</width>
      <height>21</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>NONE</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>PNORM</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>TUKEY</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>HUBER</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>MEDIAN</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>DIS_TUKEY</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>FAIR</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>LOGISTIC</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>TRIMMED</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>DIS_PNORM</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>TANH</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>SET</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>1380</x>
      <y>470</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Selected Mode</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Mode_Add_ONE">
    <property name="geometry">
     <rect>
      <x>1620</x>
      <y>520</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>添加/修改</string>
    </property>
   </widget>
   <widget class="QPushButton" name="MODE_ADD_TWO">
    <property name="geometry">
     <rect>
      <x>1620</x>
      <y>610</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>添加/修改</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>1540</x>
      <y>470</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Order</string>
    </property>
   </widget>
   <widget class="QVTKWidget" name="CloudViewer">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>1281</width>
      <height>701</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="open_txt">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>780</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>open txt</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>730</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Set deviation</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="deviation_keep">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>770</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>keep</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="deviation_random">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>840</x>
      <y>770</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>random</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLineEdit" name="deviation_set_tough">
    <property name="geometry">
     <rect>
      <x>840</x>
      <y>820</y>
      <width>113</width>
      <height>25</height>
     </rect>
    </property>
   </widget>
   <widget class="QCheckBox" name="deviation_set">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>820</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>set</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="deviation_set_fine">
    <property name="geometry">
     <rect>
      <x>840</x>
      <y>860</y>
      <width>113</width>
      <height>25</height>
     </rect>
    </property>
   </widget>
   <widget class="QComboBox" name="Translation_Estimate">
    <property name="geometry">
     <rect>
      <x>1500</x>
      <y>110</y>
      <width>86</width>
      <height>25</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <item>
     <property name="text">
      <string>NONE</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>ICP</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>PTP</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>DSPR</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>SAICP</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>PSO</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>PTL</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Combin</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>RAICP</string>
     </property>
    </item>
   </widget>
   <widget class="QComboBox" name="Correspond">
    <property name="geometry">
     <rect>
      <x>1460</x>
      <y>360</y>
      <width>151</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>NONE</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Trandition</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Direction Constraint</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Fit Plane</string>
     </property>
    </item>
   </widget>
   <widget class="QLineEdit" name="Paraments_Value">
    <property name="geometry">
     <rect>
      <x>1550</x>
      <y>210</y>
      <width>113</width>
      <height>25</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>1540</x>
      <y>530</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Left</string>
    </property>
   </widget>
   <widget class="QLabel" name="Mode_Add_2">
    <property name="geometry">
     <rect>
      <x>1540</x>
      <y>610</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Right</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Reset_Mode">
    <property name="geometry">
     <rect>
      <x>1620</x>
      <y>10</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>重置</string>
    </property>
   </widget>
   <widget class="QPushButton" name="save_txt">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>820</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>save txt</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="Selected_Mode_1">
    <property name="geometry">
     <rect>
      <x>1380</x>
      <y>510</y>
      <width>104</width>
      <height>91</height>
     </rect>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Ubuntu'; font-size:11pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;ICP&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;MEDIAN&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;0&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Trandition&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="Selected_Mode_2">
    <property name="geometry">
     <rect>
      <x>1380</x>
      <y>610</y>
      <width>104</width>
      <height>91</height>
     </rect>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Ubuntu'; font-size:11pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;ICP&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;HUBER&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;0.75&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Trandition&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QPushButton" name="add_point">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>790</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>add point</string>
    </property>
   </widget>
   <widget class="QPushButton" name="select_point">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>740</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>select point</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="center_rotation">
    <property name="geometry">
     <rect>
      <x>1200</x>
      <y>750</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>center rotation</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="random_transform">
    <property name="geometry">
     <rect>
      <x>1200</x>
      <y>840</y>
      <width>151</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>random transform</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLineEdit" name="xc_angle">
    <property name="geometry">
     <rect>
      <x>1510</x>
      <y>710</y>
      <width>113</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>0.0</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>1410</x>
      <y>720</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>x轴角度</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>1410</x>
      <y>750</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>y轴角度</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>1410</x>
      <y>780</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>z轴角度</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="yc_angle">
    <property name="geometry">
     <rect>
      <x>1510</x>
      <y>740</y>
      <width>113</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>0.0</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="zc_angle">
    <property name="geometry">
     <rect>
      <x>1510</x>
      <y>770</y>
      <width>113</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>40.0</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>1400</x>
      <y>840</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>y轴角度</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>1400</x>
      <y>810</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>x轴角度</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>1400</x>
      <y>870</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>z轴角度</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="yr_angle">
    <property name="geometry">
     <rect>
      <x>1490</x>
      <y>840</y>
      <width>113</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>60.0</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="zr_angle">
    <property name="geometry">
     <rect>
      <x>1490</x>
      <y>870</y>
      <width>113</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>90.0</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="xr_angle">
    <property name="geometry">
     <rect>
      <x>1490</x>
      <y>810</y>
      <width>113</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>30.0</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_15">
    <property name="geometry">
     <rect>
      <x>1650</x>
      <y>840</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>y轴平移</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_16">
    <property name="geometry">
     <rect>
      <x>1650</x>
      <y>810</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>x轴平移</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_17">
    <property name="geometry">
     <rect>
      <x>1650</x>
      <y>870</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>z轴平移</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="yr_distance">
    <property name="geometry">
     <rect>
      <x>1730</x>
      <y>840</y>
      <width>113</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>4.0</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="zr_distance">
    <property name="geometry">
     <rect>
      <x>1730</x>
      <y>870</y>
      <width>113</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>6.0</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="xr_distance">
    <property name="geometry">
     <rect>
      <x>1730</x>
      <y>810</y>
      <width>113</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>2.0</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_18">
    <property name="geometry">
     <rect>
      <x>1210</x>
      <y>930</y>
      <width>91</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>sample rate</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="sample_rate">
    <property name="geometry">
     <rect>
      <x>1320</x>
      <y>930</y>
      <width>113</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>1.0</string>
    </property>
   </widget>
   <zorder>CloudViewer</zorder>
   <zorder>change_point</zorder>
   <zorder>open_source</zorder>
   <zorder>insert_point</zorder>
   <zorder>delete_point</zorder>
   <zorder>Start</zorder>
   <zorder>label</zorder>
   <zorder>label_2</zorder>
   <zorder>label_3</zorder>
   <zorder>label_5</zorder>
   <zorder>Paraments</zorder>
   <zorder>label_4</zorder>
   <zorder>Mode_Add_ONE</zorder>
   <zorder>MODE_ADD_TWO</zorder>
   <zorder>label_6</zorder>
   <zorder>open_txt</zorder>
   <zorder>label_7</zorder>
   <zorder>deviation_keep</zorder>
   <zorder>deviation_random</zorder>
   <zorder>deviation_set_tough</zorder>
   <zorder>deviation_set</zorder>
   <zorder>deviation_set_fine</zorder>
   <zorder>Translation_Estimate</zorder>
   <zorder>Correspond</zorder>
   <zorder>Paraments_Value</zorder>
   <zorder>label_8</zorder>
   <zorder>Mode_Add_2</zorder>
   <zorder>Reset_Mode</zorder>
   <zorder>save_txt</zorder>
   <zorder>Selected_Mode_1</zorder>
   <zorder>Selected_Mode_2</zorder>
   <zorder>add_point</zorder>
   <zorder>select_point</zorder>
   <zorder>center_rotation</zorder>
   <zorder>random_transform</zorder>
   <zorder>xc_angle</zorder>
   <zorder>label_9</zorder>
   <zorder>label_10</zorder>
   <zorder>label_11</zorder>
   <zorder>yc_angle</zorder>
   <zorder>zc_angle</zorder>
   <zorder>label_12</zorder>
   <zorder>label_13</zorder>
   <zorder>label_14</zorder>
   <zorder>yr_angle</zorder>
   <zorder>zr_angle</zorder>
   <zorder>xr_angle</zorder>
   <zorder>label_15</zorder>
   <zorder>label_16</zorder>
   <zorder>label_17</zorder>
   <zorder>yr_distance</zorder>
   <zorder>zr_distance</zorder>
   <zorder>xr_distance</zorder>
   <zorder>label_18</zorder>
   <zorder>sample_rate</zorder>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1857</width>
     <height>28</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QVTKWidget</class>
   <extends>QWidget</extends>
   <header>QVTKWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
